src/parser.ts

```diff
@@ -2601,13 +2601,16 @@ export class Parser extends DiagnosticEmitter {
   ): ImportStatement | null {
 
     // at 'import':
-    //  ('{' (ImportMember (',' ImportMember)* '}') | ('*' 'as' Identifier)?
+    //  'type'? ('{' (ImportMember (',' ImportMember)* '}') | ('*' 'as' Identifier)?
     //  'from' StringLiteral ';'?
 
     var startPos = tn.tokenPos;
     var members: ImportDeclaration[] | null = null;
     var namespaceName: IdentifierExpression | null = null;
     var skipFrom = false;
+
+    tn.skip(Token.TYPE);
```

Comment:
In its current form, I believe we need a warning here since an `import` and an `import type` statement have different semantics. If it is a normal `import`, the respective file's top-level statements run, while if it is an `import type`, just the type is imported without actually including the file (or running it). Perhaps a `Not implemented: Type imports`?

There is also the case of `import type from "file"` that I think would currently swallow the identifier and needs to be handled with lookahead.